### PR TITLE
Add delete button for uploaded files

### LINKED ISSUE #1367: Add ability to delete files
hi
When i upload a file in message box and i want to delete it, file deleted in message box but it remain on file list!

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/28115792-add-ability-to-delete-files?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F16890630&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Burstaholic**
This will allow you to clean up file clutter :)

Fixes #1367

### CODE REVIEW

File: packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee [hunk 1]
```diff
@@ -23,11 +23,40 @@ Template.uploadedFilesList.helpers
...
 Template.uploadedFilesList.events
 	'click .room-file-item': (e, t) ->
 		if $(e.currentTarget).siblings('.icon-picture').length
 			e.preventDefault()
 
+	'click .icon-trash': (e, t) ->
+		self = this
+		swal {
+			title: 'Are you sure?'
+			text: 'You will not be able to recover this file.'
```
Comment (marceloschmidt) [inline packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee]: This text should be translatable, use `TAPi18n.__('You_will_not_be_able_to_recover_this_file)` and add `You_will_not_be_able_to_recover_this_file` to `i18n/en.i18n.json`

File: packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee [hunk 2]
```diff
@@ -23,11 +23,40 @@ Template.uploadedFilesList.helpers
...
+			showCancelButton: true
+			confirmButtonColor: '#DD6B55'
+			confirmButtonText: 'Yes delete it'
+			cancelButtonText: 'Cancel'
+			closeOnConfirm: false
+			html: false
+		}, ->
+			swal
+				title: 'Deleted'
+				text: 'Your file has been deleted.'
```
Comment (marceloschmidt) [inline packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee]: This should be translatable too

File: packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee [hunk 3]
```diff
@@ -23,11 +23,40 @@ Template.uploadedFilesList.helpers
...
+
 Template.uploadedFilesList.events
 	'click .room-file-item': (e, t) ->
 		if $(e.currentTarget).siblings('.icon-picture').length
 			e.preventDefault()
 
+	'click .icon-trash': (e, t) ->
+		self = this
+		swal {
+			title: 'Are you sure?'
```
Comment (marceloschmidt) [inline packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee]: This should be translatable too

File: packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee [hunk 4]
```diff
@@ -23,11 +23,40 @@ Template.uploadedFilesList.helpers
...
+			text: 'You will not be able to recover this file.'
+			type: 'warning'
+			showCancelButton: true
+			confirmButtonColor: '#DD6B55'
+			confirmButtonText: 'Yes delete it'
```
Comment (marceloschmidt) [inline packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee]: This should be translatable too

File: packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee [hunk 5]
```diff
@@ -23,11 +23,40 @@ Template.uploadedFilesList.helpers
...
+			cancelButtonText: 'Cancel'
```
Comment (marceloschmidt) [inline packages/rocketchat-ui-flextab/flex-tab/tabs/uploadedFilesList.coffee]: This should be translatable too

**Comment by Burstaholic** [comment]
Updated and rebased.

**Comment by marceloschmidt** [comment]
Awesome! LGTM.

**Comment by rodrigok** [comment]
LGTM

**Comment by marceloschmidt** [comment]
I just had an error when trying to delete an old file in #support channel. 

image

**Comment by rodrigok** [comment]
Yes, this PR https://github.com/RocketChat/Rocket.Chat/pull/2116?ts=2 replaced `fileCollection` by `RocketChat.models.Uploads`

**Comment by rodrigok** [comment]
I will fix that

**Comment by rodrigok** [comment]
Fixed via https://github.com/RocketChat/Rocket.Chat/commit/d3cfeb5ba4d0065e662286be56cca30763beba64?ts=2

**Pull request merged**